src/main/java/org/truffleruby/core/range/RangeNodes.java

```diff
@@ -76,6 +79,19 @@ public DynamicObject map(DynamicObject range, DynamicObject block,
             return createArray(arrayBuilder.finish(store, length), length);
         }
 
+        @Fallback
+        public Object mapFallback(Object range, Object block) {
+            return FAILURE;
+        }
+
+        private Object dispatch(DynamicObject block, Object... argumentsObjects) {
+            if (yieldNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                yieldNode = insert(new YieldNode());
+            }
+
+            return yieldNode.dispatch(block, argumentsObjects);
+        }
```

Comment:
You could use `@Cached("new()")` here to make it much shorter.

Reply:
Thanks. I didn't realize that would work. And all other usages of `YieldNode` make it a field.